### ISSUE TITLE
DockerDesktop won't start after upgraded to Windows10 build 18363

### ISSUE THREAD

**Issue by fuyili** [docker/for-win#6795]
Crash report id: 51E4D59C-9BD7-47B2-B175-8F96C76B9FE2/20200519141921

  - [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID:51E4D59C-9BD7-47B2-B175-8F96C76B9FE2/20200519141921

### Expected behavior
my docker desk top worked fine with windows 10 build 17xxx for almost a year.

### Actual behavior
After upgrade to Windows 10 build 18363, docker desktop won't start. I uninstalled and reinstalled docker a couple  times and still the same error.
Docker.Core.HttpBadResponseException:
Unhandled exception: Sequence contains no elements
   at Docker.Core.Logging.ClientExceptionInterceptor.<InterceptResponseAsync>d__0.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.Core\Logging\ClientExceptionInterceptor.cs:line 17
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Docker.Core.Logging.LoggingMessageHandler.<SendAsync>d__4.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.Core\Logging\LoggingMessageHandler.cs:line 37
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Net.Http.HttpClient.<FinishSendAsyncBuffered>d__58.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.Core.BackendAPI.BackendAPIClient.<CreateHyperVVMAsync>d__10.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.Core\BackendAPI\BackendAPIClient.cs:line 93
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.Engines.LinuxHyperVEngine.<DoStartAsync>d__12.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.Desktop\Engines\LinuxHyperVEngine.cs:line 57
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.ApiServices.StateMachines.TaskExtensions.<WrapAsyncInCancellationException>d__0.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\TaskExtensions.cs:line 29
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__5.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:line 67
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__5.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:line 92
Original stacktrace:
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source)
   at Docker.Backend.HyperVInfrastructure.<CreateDefaultVMAsync>d__12.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.Backend\HyperV\HyperVInfrastructure.cs:line 117
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.Backend.HyperVInfrastructure.<CreateOrConfigureDockerVMAsync>d__10.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.Backend\HyperV\HyperVInfrastructure.cs:line 97
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.Backend.HyperV.<CreateOrConfigureAsync>d__8.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.Backend\HyperV\HyperV.cs:line 130
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.Backend.HttpAPI.HyperVAPIController.<CreateVMAsync>d__7.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.Backend\HttpAPI\HyperVAPIController.cs:line 35
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Threading.Tasks.TaskHelpersExtensions.<CastToObject>d__0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Controllers.ApiControllerActionInvoker.<InvokeActionAsyncCore>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Controllers.ActionFilterResult.<ExecuteAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Dispatcher.HttpControllerDispatcher.<SendAsync>d__15.MoveNext()

### Information

  - Windows Version: Windows 10, build 18363
  - Docker Desktop Version: 2.3.0.2
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: No.

### Steps to reproduce the behavior

  1. ... start docker desktop from windows startup
  2. ...

**Comment by paulohpf** [comment]
i have the same issue and had everything and no success

**Comment by stoto34** [comment]
Same issue here. I remove docker desktop and install it again, no success...

**Comment by paulohpf** [comment]
I've been discovered whats going on here!
I've installed VirtualBox and Docker in the same machine... They can't co-exist on the same machine

**Comment by komalbarun** [comment]
> I've been discovered whats going on here!
> I've installed VirtualBox and Docker in the same machine... They can't co-exist on the same machine

Docker and VirtualBox can coexist on the same machine. Its HyperV and VirtualBox which cannot co-exist.

**Comment by paulohpf** [comment]
> > I've been discovered whats going on here!
> > I've installed VirtualBox and Docker in the same machine... They can't co-exist on the same machine
> 
> Docker and VirtualBox can co-exist on the same machine. Its HyperV and VirtualBox which cannot co-exist.

I understand, but can you explain better why this is happening?

**Comment by komalbarun** [comment]
> > > I've been discovered whats going on here!
> > > I've installed VirtualBox and Docker in the same machine... They can't co-exist on the same machine
> > 
> > 
> > Docker and VirtualBox can co-exist on the same machine. Its HyperV and VirtualBox which cannot co-exist.
> 
> I understand, but can you explain better why this is happening?

In my case I updated graphics drivers to fix.

Another way might be to remove the DockerVirtualVM from Hyper-V and then start docker ( **I have not tried this though** ).

**Comment by haiqun** [comment]
i have the same issue and had everything and no success

**Comment by komalbarun** [comment]
I also disabled docker startup on boot.

**Comment by mikereborn** [comment]
I was having the same issue as noted above. On the crash report, when I looked at the log file, I notice the issue was directly after it tried to boot into linux containers. So I downgraded to version 2.2.0.5 (it was hard for me to find it, so if you are like me, go here: [https://docs.docker.com/docker-for-windows/release-notes/#docker-desktop-community-2205]

Then on install I checked the box for "Use Windows Containers..."

So far, its working for me.
